x-pack/legacy/plugins/ml/public/jobs/jobs_list/components/utils.js

```diff
@@ -154,11 +155,24 @@ export function cloneJob(jobId) {
         // use tempJobCloningObjects to temporarily store the job
         mlJobService.tempJobCloningObjects.job = job;
 
-        if (job.data_counts.earliest_record_timestamp !== undefined && job.data_counts.latest_record_timestamp !== undefined) {
+        if (
+          job.data_counts.earliest_record_timestamp !== undefined &&
+          job.data_counts.latest_record_timestamp !== undefined &&
+          job.data_counts.latest_bucket_timestamp !== undefined) {
           // if the job has run before, use the earliest and latest record timestamp
           // as the cloned job's time range
+          let end = job.data_counts.latest_record_timestamp;
+
+          if (job.datafeed_config.aggregations !== undefined) {
+            // if the datafeed uses aggregations the latest_record_timestamp may not be the same
+            // as the last data point in the index, so instead use latest_bucket_timestamp and add two
+            // bucket spans minus one ms
+            const bucketSpanMs = parseInterval(job.analysis_config.bucket_span).asMilliseconds();
+            end = job.data_counts.latest_bucket_timestamp + (bucketSpanMs * 2) - 1;
+          }
+
           mlJobService.tempJobCloningObjects.start = job.data_counts.earliest_record_timestamp;
```

Comment:
I think we may need to do something to the `start` time too. With a single metric farequote job, 15m bucket span, after two clones, the `earliest_record_timestamp` has crept forward over 3 minutes:

![image](https://user-images.githubusercontent.com/7405507/65610026-a0653d80-dfa8-11e9-9a54-7a93a75a8f95.png)

Reply:
Yes, good point.  The `start` time should be rounded down to the beginning of the bucket the reported `earliest_record_timestamp` is in.  (It gets reported as the latest timestamp in the first bucket when the datafeed is using aggregations.)

Reply:
Fixed in 175db6cd269da581cb6a550d1ba6ebd336a6130a